acceptance/tests/resource/service/ticket_4123_should_list_all_running_redhat.sh

```diff
@@ -7,20 +7,20 @@
 RALSH_FILE=/tmp/ralsh-running-list-$$
 SERVICE_FILE=/tmp/service-running-list-$$
 
-puppet resource service | egrep -B1 "ensure.*=>.*'running" | grep 'service {' | gawk -F"'" '{print $2}' | sort  > $RALSH_FILE
+puppet resource service | egrep -B1 "ensure.*=>.*'running" | grep 'service {' | gawk -F"'" '{print $2}' | sort  > "$RALSH_FILE"
 
-if [ -e $SERVICE_FILE ]; then
-  rm $SERVICE_FILE
+if [ -e "$SERVICE_FILE" ]; then
+  rm "$SERVICE_FILE"
 fi
 
 SERVICEDIR='/etc/init.d'
-for SERVICE in $( ls $SERVICEDIR | sort | egrep -v "(functions|halt|killall|single|linuxconf)" ) ; do
+for SERVICE in "$( ls "$SERVICEDIR" | sort | egrep -v "(functions|halt|killall|single|linuxconf)" )" ; do
```

Comment:
You can't put double quotes around the $() call as it turns the output into one value to set SERVICE to, instead of looping over the file names and setting SERVICE to each file. We still won't protect against files with spaces in the name, but this is a test so we're probably fine

Reply:
D'oh! Fixed.